### PR TITLE
Reset the noEmitForJsFiles option when updating compiler options

### PR DESCRIPTION
**Pull request by zhengbli**

### CODE REVIEW

**Comment by vladima** [comment]
Can you add tests?

File: src/server/project.ts [hunk 1]
```diff
@@ -266,6 +264,12 @@ namespace ts.server {
             this.markAsDirty();
         }
 
+        setInternalCompilerOptionsForEmittingJsFiles() {
```
Comment (vladima) [inline src/server/project.ts]: `private`

**Comment by billti** [comment]
Thanks @zhengbli . If we put this in the next VS RC, we'll need just this change off of the current commit in the package. Can you create a new branch (e.g. `vs/release-2.1`) off of commit https://github.com/Microsoft/TypeScript/commit/5108285 and apply just this change?

**Comment by zhengbli** [comment]
Sure, working on it

**Comment by zhengbli** [comment]
@billti I created https://github.com/Microsoft/TypeScript/tree/vs/release-2.1 that has the changes in this PR. Thank you!

**Pull request merged**